rasa_core/broker.py

```diff
@@ -52,3 +89,45 @@ def _publish(self, body):
 
     def _close(self):
         self.connection.close()
+
+
+class FileProducer(EventChannel):
+    """Log events to a file in json format.
+
+    There will be one event per line and each event is stored as json."""
+
+    DEFAULT_LOG_FILE_NAME = "rasa_event.log"
+
+    def __init__(self, path=None) -> None:
+        self.path = path
+        self.event_logger = self._event_logger(path)
+
+    @classmethod
+    def from_endpoint_config(cls, broker_config) -> Optional['FileProducer']:
+        if broker_config is None:
+            return None
+
+        # noinspection PyArgumentList
+        return cls(**broker_config.kwargs)
+
+    def _event_logger(self, file_name):
+        """Instantiate the file logger"""
+
+        logger_file = file_name or self.DEFAULT_LOG_FILE_NAME
+        # noinspection PyTypeChecker
+        query_logger = logging.getLogger('event-logger')
+        query_logger.setLevel(logging.INFO)
+        ch = logging.FileHandler(logger_file)
```

Comment:
use encoding parameter + rename `ch` to sth like `handler` or so

Reply:
do we need file rotating?

Reply:
at some point: yes. but let's do that another time

Reply:
:+1: